engine/src/lib.rs

```diff
@@ -6,14 +6,17 @@ use std::collections::HashMap;
 use std::num::NonZeroUsize;
 use std::path::Path;
 
+pub mod gas;
 pub mod sync;
 #[cfg(test)]
 mod tests;
 pub mod tracing;
 pub mod types;
 
+pub type SharedStorage = std::sync::Arc<tokio::sync::RwLock<Storage>>;
+
 pub struct EngineContext {
-    pub storage: Storage,
+    pub storage: SharedStorage,
```

Comment:
Mostly thinking out loud here, but I wonder if there is a way to make this a generic type parameter. This is a general library and not all users of the library may require the shared storage. I know it is now needed in `refiner-app`, but if it's not needed in another application then having the `Arc` and `RwLock` on top adds unnecessary overhead. If we could somehow generalize this so that either it could be async-friendly or not then we could have the best of both worlds.

Reply:
Left out for a future improvement.